projects/shr-mediator/src/botswana/controllers/kafka-consumer.controller.ts

```diff
@@ -38,11 +38,13 @@ export class KafkaConsumerController {
 
   @EventPattern(topicList.SEND_ADT_TO_IPMS)
   async handleSendAdtToIpms(
-    @Payload() val: string,
+    @Payload() val: IBundle,
     @Ctx() context: KafkaContext,
   ) {
     try {
-      const origBundle = JSON.parse(val).bundle;
+
+      //@ts-ignore
+      const origBundle: IBundle = val.bundle;
```

Comment:
We need to avoid @ts-ignore, I think we can fix `@Payload() val: IBundle,` so that it returns the correct type

Reply:
export interface BundlePayload {
  bundle: IBundle;
}
@Payload() val: BundlePayload